rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt

```diff
@@ -703,6 +703,19 @@ open class InternalInterpreter(
         val quotient = dividend.divide(divisor, MathContext.DECIMAL128)
         val type = statement.target.variable.referred!!.type
         require(type is NumberType)
+        // calculation of rest
+        // NB. rest based on type of quotient
+        if (statement.mvrTarget != null) {
+            val restType: NumberType = if (statement.mvrTarget.type() is NumberType) {
```

Comment:
`require` function allows both to check and to cast, you can replace the statements as below:

```
val restType = statement.mvrTarget.type()
require(restType is NumberType)
```

Reply:
Fixed